### DISCUSSION TITLE
What should be the set of files/folders in the base template

### DISCUSSION THREAD

**Discussion by brumhard** [SchwarzIT/go-template#72] [Q&A]
Hi there,
currently I feel like the set of files/folders included in the base template could be a bit too big.
If personally basiscally always remove at least the `/deployments`, `/assets`, `internal` and `configs`. Also you could argue that the Dockerfile is not needed in every Go repo.

Therefore maybe it would be worth thinking about removing some parts of the base template completely or make them optional.

Maybe it would also be nice to include a `--lean` flag to really only include config files and no folders at all. But that could potentially be another discussion.

---
For context: there have been several personal requests to restructure a bit and there was also some discussion in the [release reddit post](https://www.reddit.com/r/golang/comments/q8oa37/gotemplate_bootstrap_you_next_go_project_with_ease/hgt46xm/)

**Comment by brumhard** [comment]
@dergeberl @linuxluigi @danielzwink any opinions? :)

**Reply by linuxluigi** [reply]
I think this would be a very good starting point to write the documrntation for the template and write down, what we are thinking for all the folders and files. Then we will figure out at best, which folder stay and which should go.

About the dockerfile, this is a bigger topic. Before we make this file optional, clean up the makefile. This file is to full in my opinion.

**Comment by danielzwink** [comment]
Unfortunately I hardly use the template due to lack of coding practice and therefore a strong opinion is difficult. But I tend to leave some opinion in the template and therefore also leave the folders. At least `/assets`, `/deploments` and `/internal` seem to me being useful defaults which are also quickly deleted.

**Comment by brumhard** [comment]
After some more discussion we decided to use #106 as a first step to cleanup the template.
In this we simplify the structure by removing `/assets`, `configs`, `deployments`, `docs`.
We agreed that these are too rarely used and while it's easy to remove them if they are not needed it's just as easy the just add them when they are needed.
Also, `/assets`, `configs` and `deployments` are often not used or named differently (e.g. `charts` instead of deployments, configs in the root directly).

This is our first step to a cleaner more versatile template.
